geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/DescriptiveStats.scala

```diff
@@ -0,0 +1,358 @@
+
+  implicit def toDenseMatrix64F(sm: SimpleMatrix): DenseMatrix64F = sm.getMatrix
+
+  implicit class SimpleMatrixOps(a: SimpleMatrix) {
+
+    def +(b: Double): SimpleMatrix = a.plus(b)
+    def +(b: SimpleMatrix): SimpleMatrix = a.plus(b)
+
+    def +=(b: Double): Unit = CommonOps.add(a, b, a)
+    def +=(b: SimpleMatrix): Unit = CommonOps.add(a, b, a)
+
+    def -(b: Double): SimpleMatrix = a.minus(b)
+    def -(b: SimpleMatrix): SimpleMatrix = a.minus(b)
+
+    def -=(b: Double): Unit = CommonOps.subtract(a, b, a)
+    def -=(b: SimpleMatrix): Unit = CommonOps.subtract(a, b, a)
+
+    def *(b: Double): SimpleMatrix = a.scale(b)
+    def *(b: SimpleMatrix): SimpleMatrix = a.elementMult(b)
+
+    def *=(b: Double): Unit = CommonOps.scale(b, a, a)
+    def *=(b: SimpleMatrix): Unit = CommonOps.elementMult(a, b, a)
+
+    def /(b: Double): SimpleMatrix = a.divide(b)
+    def /(b: SimpleMatrix): SimpleMatrix = a.elementDiv(b)
+
+    def /=(b: Double): Unit = CommonOps.divide(a, b, a)
+    def /=(b: SimpleMatrix): Unit = CommonOps.elementDiv(a, b, a)
+
+    def **(b: Double): SimpleMatrix = a.elementPower(b)
+    def **=(b: Double): Unit = CommonOps.elementPower(a, b, a)
+
+    def diag(v: Double): SimpleMatrix = {
+      val m = new SimpleMatrix(a)
+      (0 until Math.min(m.getNumRows, m.getNumCols)).foreach(i => m.set(i, i, v))
+      m
+    }
+
+    def |*|(b: SimpleMatrix): SimpleMatrix = a.mult(b)
+
+    def T: SimpleMatrix = a.transpose
+
+    def min(values: SimpleMatrix): Unit = {
+      val n = values.getNumElements
+      var i = 0
+      while (i < n) {
+        val v = values.get(i)
+        val min = a.get(i)
+        if (v < min) {
+          a.set(i, v)
+        }
+        i += 1
+      }
+    }
+    def max(values: SimpleMatrix): Unit = {
+      val n = values.getNumElements
+      var i = 0
+      while (i < n) {
+        val v = values.get(i)
+        val max = a.get(i)
+        if (v > max) {
+          a.set(i, v)
+        }
+        i += 1
+      }
+    }
+  }
+
+  implicit class DoubleOps(a: Double) {
+    def +(b: SimpleMatrix): SimpleMatrix = b.plus(a)
+    def -(b: SimpleMatrix): SimpleMatrix = {
+      val c = new SimpleMatrix(b.getNumRows, b.getNumCols)
+      CommonOps.subtract(a, b, c)
+      c
+    }
+    def *(b: SimpleMatrix): SimpleMatrix = b.scale(a)
+    def /(b: SimpleMatrix): SimpleMatrix = {
+      val c = new SimpleMatrix(b.getNumRows, b.getNumCols)
+      CommonOps.divide(a, b, c)
+      c
+    }
+    def **(b: SimpleMatrix): SimpleMatrix = {
+      val c = new SimpleMatrix(b.getNumRows, b.getNumCols)
+      CommonOps.elementPower(a, b, c)
+      c
+    }
+  }
+
+  def updateMinMax(mins: SimpleMatrix, maxs: SimpleMatrix, values: SimpleMatrix): Unit = {
+    val n = values.getNumElements
+    var i = 0
+    while (i < n) {
+      val v = values.get(i)
+      val min = mins.get(i)
+      val max = maxs.get(i)
+      if (v < min) {
+        mins.set(i, v)
+      } else if (v > max) {
```

Comment:
since min starts at Double.MaxValue, and max starts at negative max value, should this be two separate ifs instead of an else if? Seems like if we evaluated e.g. [10, 9, 8, 7, 6] max would never get updated.